app/static/js/page.js

```
@@ -727,18 +727,88 @@ function snakeOilApp() {
-    hasWikipediaLink(entityType, entity) {
+    hasEntityLink(entityType, entity) {
         if (!this.wikipediaData || !this.wikipediaData[entityType]) return false;
-        return this.wikipediaData[entityType][entity] !== null && this.wikipediaData[entityType][entity] !== undefined;
+        const entityData = this.wikipediaData[entityType][entity];
+        return entityData !== null && entityData !== undefined && entityData.link;
     },
     
-    getWikipediaData(entityType, entity) {
-        if (!this.hasWikipediaLink(entityType, entity)) return null;
+    // Legacy function for backward compatibility
+    hasWikipediaLink(entityType, entity) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)